staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go

```diff
@@ -196,6 +206,16 @@ func (s *ServerRunOptions) Validate() []error {
 	if err := validateCorsAllowedOriginList(s.CorsAllowedOriginList); err != nil {
 		errors = append(errors, err)
 	}
+	if s.FeatureGate != nil {
+		if errs := s.FeatureGate.Validate(); len(errs) != 0 {
+			errors = append(errors, errs...)
+		}
+	}
+	if s.EffectiveVersion != nil {
```

Comment:
instead of allowing nil, can we fail on nil? We're requiring this on construction and really every project should make a choice.